src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStatementOfTruthPaySubmit.java

```diff
@@ -93,21 +96,30 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
         final CaseData caseData = details.getData();
         final State currentState = details.getState();
 
+        updateRespondentDigitalDetails(caseData);
+
         if (!caseData.hasStatementOfTruth() || !caseData.hasSolSignStatementOfTruth()) {
 
             return AboutToStartOrSubmitResponse.<CaseData, State>builder()
                 .data(caseData)
                 .state(currentState)
-                .errors(asList("Statement of truth for solicitor and applicant 1 needs to be accepted"))
+                .errors(singletonList("Statement of truth for solicitor and applicant 1 needs to be accepted"))
                 .build();
         }
 
-        final State resultState = solicitorSubmitPetitionService.aboutToSubmit(caseData, details.getId());
+        return solicitorSubmitPetitionService.aboutToSubmit(caseData, details.getId());
 
-        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
-            .data(caseData)
-            .state(resultState)
-            .build();
+    }
+
+    private void updateRespondentDigitalDetails(CaseData caseData) {
+        if (null != caseData.getRespondentOrganisationPolicy()) {
+            String respondentOrgId = caseData.getRespondentOrganisationPolicy().getOrganisation().getOrganisationId();
+            if (YES.equals(caseData.getRespSolDigital()) && !Strings.isNullOrEmpty(respondentOrgId)) {
```

Comment:
Could these be moved into CaseData as methods isRespSolDigital and hasRespondentOrgId?